langkit/expressions/base.py

```diff
@@ -3212,6 +3212,21 @@ def __init__(self, expr, prefix, name=None, doc=None, public=None,
         :type: bool
         """
 
+        self.is_artificial_dispatcher = False
```

Comment:
I remember feeling uneasy about this at some point when I worked on trying to unify the 17 fields we have for the names of properties, but anyway, for me this is really not so clear: 

All dispatchers are generated, insofar as the code is generated, so saying that some dispatchers "come from source" is very misleading IMO. What this means is that you didn't have to move the true body of the property to another property because there is no body to speak of.

Also it seems like the only use of this field is for your assertion in dsl_unparse ?

Anyway, I wonder if a simpler solution to that problem might just be running the unparser *before* those expansion passes ?

Reply:
I’d prefer avoiding moving the dsl_unparse pass so that it’s still in the “emit” pass and thus so that we can rely on the fact that all compilation happened (i.e. all information is available).

As discussed live, the dispatch lowering pass is very tricky, and there is a mind-twisting reuse of PropertyDef objects, making it hard to think about object identity across the compilation pipeline. I’ll try to simplify this first and remove the need for this extra field.

Reply:
> As discussed live, the dispatch lowering pass is very tricky, and there is a mind-twisting reuse of PropertyDef objects, making it hard to think about object identity across the compilation pipeline. I’ll try to simplify this first and remove the need for this extra field.

Ok, that works for me, although it's definitely not urgent so keep that for later if you prefer!